### PR TITLE
Moves collapse animation to FlatList, updates timings

### PR DESCRIPTION
**Pull request by shazm**
#### Summary
Removes the animations from the `ChannelItem` and pushes them on to the `FlatList` inside `CategoryBody` further up the tree.

It also updates the timings to `200` / `300` for the opacity easing in and out for a better experience.

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [X] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Review by enahum — APPROVED** [review]
Nice one!

File: app/screens/home/channel_list/categories_list/categories/body/category_body.tsx [hunk 1]
```diff
@@ -39,26 +40,40 @@ const CategoryBody = ({sortedChannels, category, hiddenChannelIds, limit, onChan
         return (
             <ChannelItem
                 channel={item}
-                collapsed={category.collapsed}
                 testID={`category.${category.displayName.replace(/ /g, '_').toLocaleLowerCase()}.channel_list_item`}
                 onPress={onChannelSwitch}
             />
         );
     }, [category.collapsed, onChannelSwitch]);
```
Comment (larkox) [inline app/screens/home/channel_list/categories_list/categories/body/category_body.tsx:47]: No need to depend on category.collapsed anymore apparently.

**Review by larkox — APPROVED** [review]
LGTM, just one small change.

File: app/screens/home/channel_list/categories_list/categories/body/category_body.tsx [hunk 2]
```diff
@@ -39,26 +40,40 @@ const CategoryBody = ({sortedChannels, category, hiddenChannelIds, limit, onChan
...
-    }, [category.collapsed, onChannelSwitch]);
+    }, [onChannelSwitch]);
+
+    const sharedValue = useSharedValue(category.collapsed);
+
+    useEffect(() => {
+        sharedValue.value = category.collapsed;
+    }, [category.collapsed]);
+
+    const height = ids.length ? ids.length * 40 : 0;
```
Comment (enahum) [inline app/screens/home/channel_list/categories_list/categories/body/category_body.tsx:55]: @shazm there is an issue here where we are not taking into account channels that have been archived and are not the active channel as well as the ones that should not render as they are in the unread section, basically by not taken them into account there is additional space that should not be there.

**Review by enahum — CHANGES_REQUESTED** [review]

**Review by matthewbirtch — APPROVED** [review]
Seems good to me now with your latest changes @enahum

**Review by enahum — APPROVED** [review]

**Pull request merged**